### PR TITLE
!!!EXTREME!!! Mob refactor, with fixes

### PR DESCRIPTION
**Pull request by Crazylemon64**
Code-wise, this does:
Refactor mob status effects/health so as to allow instant blindness/etc. updates
Refactors mob `stat` to prevent accidentally reviving a mob - use `KnockOut` and `WakeUp` to make a mob conscious/unconscious.
Make all revival use `update_revive` to make sure all book-keeping checks out.
TODO:
- [x] Make sleeping and unconsciousness apply the blindness overlays
- [x] Make sure all status effects tick down appropriately
- [ ] Make sure all status updates are instant
- [ ] Ensure all animations still play correctly

:cl:Crazylemon
tweak: Sight modifications (welding helmets, blinding,etc.) are now instant, instead of being tied to "Life".
tweak: Other status modifications (stun, sleep, damage, etc.) are also now instant.
/:cl:

### CODE REVIEW

**Comment by Lady-Luck** [comment]
>Refactor mob sight so as to allow instant blindness/etc. updates

This actually led me to implicating someone as a changeling because of a particularly huge delay on blindness from a blindness sting. I felt really bad about it post-round when I learned just how finicky vision checks were.

**Comment by Fox-McCloud** [comment]
If we're being totally honest with ourselves, eventually we'll *probably* do TG's life refactor (considering how much it saves in CPU). To that end, and for consistency, I would use the same proc names TG does for blindness/blurry eyes: https://github.com/tgstation/tgstation/blob/master/code/modules/mob/status_procs.dm#L156

**Comment by Fox-McCloud** [comment]
I doubt you're crazy enough for a full vision refactor....but in case you are....

https://github.com/tgstation/tgstation/pull/15177
https://github.com/tgstation/tgstation/pull/15280
https://github.com/tgstation/tgstation/pull/15281
https://github.com/tgstation/tgstation/pull/15327

This was their vision refactor, which made *all* vision statuses update instantaneously, even if they don't involve equipping items/taking them off.

**Comment by Crazylemon64** [comment]
Guess why I'm called *crazy* lemon

**Comment by Crazylemon64** [comment]
**Their** blind and blurry eye proc names are inconsistent with the rest of the status effects - fixing those is as simple as a Find and Replace in a competent text editor, anyways.

**Comment by Lady-Luck** [comment]
Would it be possible to merge some of the more essential fixes in separate PRs? I'm assuming the full refactor will take a while.

**Comment by Crazylemon64** [comment]
MC resolved

**Comment by Crazylemon64** [comment]
Also a lot of stuff is still broken, so hold on to yer boots

**Comment by Fox-McCloud** [comment]
The fact we can't see all the changes is...wellll, problematic.

**Comment by Crazylemon64** [comment]
Darn, the refactor was too extreme

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1166,24 +1166,6 @@ var/global/list/common_tools = list(
...
-
-//check if mob is lying down on something we can operate him on.
-/proc/can_operate(mob/living/carbon/M)
-	return (locate(/obj/machinery/optable, M.loc) && M.resting) || \
-	(locate(/obj/structure/stool/bed/roller, M.loc) && 	\
-	(M.buckled || M.lying || M.weakened || M.stunned || M.paralysis || M.sleeping || M.stat)) && prob(75) || 	\
-	(locate(/obj/structure/table/, M.loc) && 	\
-	(M.lying || M.weakened || M.stunned || M.paralysis || M.sleeping || M.stat) && prob(66))
-
 /proc/reverse_direction(var/dir)
```
Comment (Crazylemon64) [inline code/__HELPERS/unsorted.dm]: Moved to code/modules/surgery/helpers.dm

**Comment by Crazylemon64** [comment]
I'll attempt to break down the changes over multiple other PRs until this is something reasonable to review

**Comment by Lady-Luck** [comment]
I love and appreciate the work you do on this github @Crazylemon64, you're my favorite coder. :heart:

**Comment by Crazylemon64** [comment]
@Lady-Luck Aww, shucks <3

**Comment by Crazylemon64** [comment]
Lessee, still tremendous, but smaller. Now, @Markolie needs the vision refactor ported for #5806, so that's next on the carving block.

**Comment by Crazylemon64** [comment]
#5840

**Comment by Crazylemon64** [comment]
this is stupidly out of date, and I had this open only to tear old chunks of code out of

it'll do the job fine even when closed

**Pull request closed without merging**